nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/kinesis/firehose/PutKinesisFirehose.java

```diff
@@ -0,0 +1,184 @@
+
+    /**
+     * Kinesis put record response error code
+     */
+    public static final String AWS_KINESIS_FIREHOSE_ERROR_CODE = "aws.kinesis.firehose.error.code";
+
+    /**
+     * Kinesis put record response record id
+     */
+    public static final String AWS_KINESIS_FIREHOSE_RECORD_ID = "aws.kinesis.firehose.record.id";
+
+    public static final List<PropertyDescriptor> properties = Collections.unmodifiableList(
+            Arrays.asList(KINESIS_FIREHOSE_DELIVERY_STREAM_NAME, MAX_BUFFER_INTERVAL,
+                  MAX_BUFFER_SIZE, BATCH_SIZE, REGION, ACCESS_KEY, SECRET_KEY, CREDENTIALS_FILE, AWS_CREDENTIALS_PROVIDER_SERVICE, TIMEOUT,
+                  PROXY_HOST,PROXY_HOST_PORT));
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    protected Collection<ValidationResult> customValidate(final ValidationContext validationContext) {
+        final List<ValidationResult> problems = new ArrayList<>(super.customValidate(validationContext));
+        final boolean batchSizeSet = validationContext.getProperty(BATCH_SIZE).isSet();
+
+        if ( batchSizeSet) {
+           int batchSize = validationContext.getProperty(BATCH_SIZE).asInteger();
+           if ( batchSize < 1 || batchSize > 500 ) {
+                problems.add(new ValidationResult.Builder().input("Batch Size").valid(false).explanation("Batch size must be between 1 and 500 but was " + batchSize).build());
+           }
+        }
+
+        final boolean maxBufferIntervalIsSet = validationContext.getProperty(MAX_BUFFER_INTERVAL).isSet();
+        if ( maxBufferIntervalIsSet) {
+           int maxBufferInterval = validationContext.getProperty(MAX_BUFFER_INTERVAL).asInteger();
+           if ( maxBufferInterval < 60 || maxBufferInterval > 900 ) {
+                problems.add(new ValidationResult.Builder().input("Max Buffer Interval").valid(false)
+                   .explanation("Max Buffer Interval must be between 60 and 900 seconds but was " + maxBufferInterval).build());
+           }
+        }
+
+        final boolean maxBufferSizeIsSet = validationContext.getProperty(MAX_BUFFER_SIZE).isSet();
+        if ( maxBufferSizeIsSet) {
+           int maxBufferSize = validationContext.getProperty(MAX_BUFFER_SIZE).asInteger();
+           if ( maxBufferSize < 1 || maxBufferSize > 128 ) {
+                problems.add(new ValidationResult.Builder().input("Max Buffer Size").valid(false).explanation("Max Buffer Size must be between 1 and 128 (mb) but was " + maxBufferSize).build());
+           }
+        }
+        return problems;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) {
+        final int batchSize = context.getProperty(BATCH_SIZE).asInteger();
+
+        List<FlowFile> flowFiles = session.get(batchSize);
+        if (flowFiles == null || flowFiles.size() == 0) {
```

Comment:
Minor:  this will never return null, only an empty collection in the case of there being no input